### PR TITLE
Fix asteroid wrapping

### PR DESCRIPTION
**Pull request by MjnMixael**
Per discussion with WookieJedi, this fixes/enhances asteroid field spawning and wrapping for certain fields.

Basically fields with "enhanced checks" do not use range to limit the player view cone when wrapping asteroids and **do** use the view cone to check if an asteroid should be spawned (where retail does not have that check at all).

The core issue is that for small fields the existing checks for wrapping asteroids from one side to the other weren't stringent enough causing asteroids to visibly pop into and out of existence. This PR fixes that issue by adding an optional parameter for asteroid fields for "enhanced checks" that makes sure the asteroid is not within view of the player when it might be wrapped as well as when an asteroid might be created.

This shouldn't be a general bugfix because it can cause fields to entirely disappear if the player stares at them long enough resulting in major mission changes. The optional parameter preserves old behavior while offering an option for new styles of fields. By having this as a parameter of `Asteroid_field` it means that this can be controlled individually for each field in a mission. Sexp compatibility is also added here, albeit in a kind of funky way to preserve backwards compatibility with set-asteroid-field's argument list.

Finally, as part of this fix I redesigned the `asteroid_maybe_reposition` function to make it more readable at a glance. Instead of doing some checks, moving the asteroid, and then maybe moving it back after more checks are completed the function does all checks up front and then only moves the asteroid around if appropriate.

### CODE REVIEW

**Comment by wookieejedi** [comment]
Tested and works as expected with both default behavior and new/fixed behavior when enabled. If anyone wants to see a video of the old buggy behavior I'd be happy to share one. 

I'll also highlight that this PR updates the dot product threshold used for 'in view' checks, which is a very needed fix. The `0.7` threshold used in the ‘in view’ checks seem to have been mostly designed for 1024x768 aspect, not 16:9 or others. So asteroids that were thought of as being out of view were actually in view. Using `cos(Proj_fov)` instead of just `0.7` fixes it and allows it to adapt to the users' actual screen. The code base was checked for similar `0.7` checks and the only time it is used for 'in view' like this is these two functions in the asteroid code and once in the sun glare code. Also noting that cosine is rather cheap and checking it every frame will allow things like Fov changes from sexps to still work as intended.

**Review by Baezon — APPROVED** [review]
Looks good, excluding any possible multi concerns.

**Comment by MjnMixael** [comment]
> Looks good, excluding any possible multi concerns.

Per discussion in SCP; as long as `Eye_position` is not null then I don't think this PR will change behavior in multi.

File: code/asteroid/asteroid.cpp [hunk 1]
```diff
@@ -990,6 +1017,32 @@ static void sanitize_asteroid_targets_list() {
...
+bool asteroid_is_within_view(vec3d *pos, float range, bool range_override)
+{
+	vec3d vec_to_asteroid;
+
+	// Distance and view cone for the position in relation to the player
+	float dist = vm_vec_normalized_dir(&vec_to_asteroid, pos, &Eye_position);
+	float dot = vm_vec_dot(&Eye_matrix.vec.fvec, &vec_to_asteroid);
+
+	// if asteroid is within view or far enough away then wrap
+	if (dot > cos(Proj_fov)) {
```
[the remark this answers is not in the document]
Comment (Baezon) [reply code/asteroid/asteroid.cpp:1033]: Since I see this was adapted from the earlier code, this is fine.

**Comment by wookieejedi** [comment]
Will plan to merge later today

**Pull request merged**